Userland/Libraries/LibGfx/ImageFormats/JPEGLoader.cpp

```diff
@@ -891,6 +891,7 @@ static ErrorOr<void> read_huffman_table(Stream& stream, JPEGLoadingContext& cont
         }
 
         table.codes.ensure_capacity(total_codes);
+        table.symbols.ensure_capacity(total_codes);
```

Comment:
Not a full review, just a quick nitpick: If we ensure we have enough capacity beforehand we can use an unchecked_append below to get rid of the vector resize checks.

Reply:
I would rather keep these checks, they don't hurt performance at all.
Actually, I did it more for readability than performance, these vectors don't contain more than a few hundreds elements.